src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -371,61 +340,104 @@ private static void writeMetadata( final ReadMetadata readMetadata,
         final Broadcast<HopscotchUniqueMultiMap<String, Integer, QNameAndInterval>> broadcastQNamesMultiMap =
                 ctx.broadcast(qNamesMultiMap);
-        final int nPartitions = reads.partitions().size();
-
-        final Map<Integer, String> intervalDispositions =
+        final List<AlignedAssemblyOrExcuse> intervalDispositions =
             reads
                 .mapPartitionsToPair(readItr ->
                         new ReadsForQNamesFinder(broadcastQNamesMultiMap.value(), nIntervals,
-                                includeMappingLocation).call(readItr).iterator(), false)
+                                includeMappingLocation, dumpFASTQs).call(readItr).iterator(), false)
                 .combineByKey(x -> x,
                                 FindBreakpointEvidenceSpark::combineLists,
                                 FindBreakpointEvidenceSpark::combineLists,
-                                new HashPartitioner(nPartitions), false, null)
-                .map(fastqHandler)
-                .collect()
-                .stream()
-                .collect(Collectors.toMap(Tuple2::_1, Tuple2::_2));
+                                new HashPartitioner(nIntervals), false, null)
+                .map(fastqHandler::apply)
+                .collect();
 
         broadcastQNamesMultiMap.destroy();
+        BwaMemIndexSingleton.closeAllDistributedInstances(ctx);
 
         return intervalDispositions;
     }
 
     /** Concatenate two lists. */
-    private static List<byte[]> combineLists( final List<byte[]> list1, final List<byte[]> list2 ) {
-        final List<byte[]> result = new ArrayList<>(list1.size() + list2.size());
+    private static List<FastqRead> combineLists( final List<FastqRead> list1, final List<FastqRead> list2 ) {
+        final List<FastqRead> result = new ArrayList<>(list1.size() + list2.size());
         result.addAll(list1);
         result.addAll(list2);
         return result;
     }
 
-    /** write a FASTQ file for an assembly */
-    @VisibleForTesting static Tuple2<Integer, String> writeFastq( final Tuple2<Integer, List<byte[]>> intervalAndFastqs,
-                                    final String outputDir, final int maxFastqSize ) {
-        final List<byte[]> fastqsList = intervalAndFastqs._2;
-        SVFastqUtils.sortFastqRecords(fastqsList);
-
-        final int fastqSize = fastqsList.stream().mapToInt(fastqRec -> fastqRec.length).sum();
-        final String disposition;
-        if ( fastqSize > maxFastqSize ) disposition = "FASTQ not written -- too big (" + fastqSize + " bytes).";
-        else {
-            final String fileName = outputDir + "/assembly" + intervalAndFastqs._1() + ".fastq";
-            SVFastqUtils.writeFastqFile(fileName, null, fastqsList);
-            disposition = fileName;
+    /** This FASTQHandler aligns assembly contigs with BWA, along with some optional writing of intermediate results. */
+    private static final class FastqWriter implements FASTQHandler {
```

Comment:
Agreed.  Name could be better.
Doesn't necessarily have anything to do with FASTQ, though.  That was just a means to an end when we were using SGA.
Maybe it's a LocalAssemblyHandler interface, and a FermiLiteAssemblyHandler implementation?